install/swiftly-install.sh

```diff
@@ -170,26 +167,198 @@ install_system_deps () {
+    docker_platform_name="amazonlinux"
+    docker_platform_version="$1"
+    package_manager="yum"
+
+    if [[ -z "$PLATFORM_NAME_PRETTY" ]]; then
+        PLATFORM_NAME_PRETTY="Amazon Linux 2"
+    fi
+}
+
+set_platform_rhel () {
+    PLATFORM_NAME="ubi$1"
+    PLATFORM_NAME_FULL="ubi$1"
+    docker_platform_name="rhel-ubi"
+    docker_platform_version="$1"
+    package_manager="yum"
+
+    if [[ -z "$PLATFORM_NAME_PRETTY" ]]; then
+        PLATFORM_NAME_PRETTY="RHEL 9"
+    fi
+}
+
+detect_platform () {
+    if [[ -f "/etc/os-release" ]]; then
+        OS_RELEASE="/etc/os-release"
+    elif [[ -f "/usr/lib/os-release" ]]; then
+        OS_RELEASE="/usr/lib/os-release"
+    else
+        manually_select_platform
+    fi
+
+    source "$OS_RELEASE"
+    PLATFORM_NAME_PRETTY="$PRETTY_NAME"
+
+    case "$ID$ID_LIKE" in
+        *"amzn"*)
+            if [[ "$VERSION_ID" != "2" ]]; then
+                manually_select_platform
+            else
+                set_platform_amazonlinux "2"
+            fi
+            ;;
+
+        *"ubuntu"*)
+            case "$UBUNTU_CODENAME" in
+                "jammy")
+                    set_platform_ubuntu "22" "04"
+                    ;;
+
+                "focal")
+                    set_platform_ubuntu "20" "04"
+                    ;;
+
+                "bionic")
+                    set_platform_ubuntu "18" "04"
+                    ;;
+
+                *)
+                    manually_select_platform
+                    ;;
+            esac
+            ;;
+
+        *"rhel"*)
+            if [[ "$VERSION_ID" != 9* ]]; then
+                manually_select_platform
+            else
+                set_platform_rhel "9"
+            fi
+            ;;
+
+        *)
+            manually_select_platform
+            ;;
+    esac
+}
+
+manually_select_platform () {
+    if [[ "$DISABLE_CONFIRMATION" == "true" ]]; then
+        echo "Error: Unsupported platform: $PRETTY_NAME"
+        exit 1
+    fi
+    echo "$PRETTY_NAME is not an officially supported platform, but the toolchains for another platform may still work on it."
+    echo ""
+    echo "Please select the platform to use for toolchain downloads:"
+
+    echo "0) Cancel"
+    echo "1) Ubuntu 22.04"
+    echo "2) Ubuntu 20.04"
+    echo "3) Ubuntu 18.04"
+    echo "4) RHEL 9"
+    echo "5) Amazon Linux 2"
+
+    read_input_with_default "0"
+    case "$READ_INPUT_RETURN" in
+        "1" | "1)")
+            set_platform_ubuntu "22" "04"
+            ;;
+
+        "2" | "2)")
+            set_platform_ubuntu "20" "04"
+            ;;
+
+        "3" | "3)")
+            set_platform_ubuntu "18" "04"
+            ;;
+
+        "4" | "4)")
+            set_platform_rhel "9"
+            ;;
+
+        "5" | "5)")
+            set_platform_amazonlinux "2"
+            ;;
+
+        *)
+            echo "Cancelling installation."
+            exit 0
+            ;;
+    esac
+}
+
+verify_getopt_install () {
```

Comment:
`getopt` is a super useful tool for parsing CLI options in bash scripts. It's included in the util-linux package, but in practice basically every Linux distro comes with it pre-installed...except Amazon Linux 2. The alternatives are manual parsing (which gets tricky with stuff like `-pubuntu20.04`, `-p  ubuntu20.04`, `--platform=ubuntu20.04`, and `--platform ubuntu20.04`) or using the shell builtin `getopts`, which doesn't support long names like `--platform`. Given that we already require `curl` to be installed, it didn't seem like a huge problem to add an additional dependency on `util-linux`.